core/themes/seven/js/script.js

```diff
@@ -0,0 +1,292 @@
+/**
+ * @file
+ * Responsive Admin tabs.
+ */
+
+(function ($) {
+  "use strict";
+
+  Backdrop.behaviors.responsivePrimaryTabs = {
+    attach: function(context, settings) {
+      var $primaryTabs = $('ul.tabs.primary');
+      var $tabsWrapper = $($primaryTabs).parent();
+      var $tabs = $('li', $primaryTabs);
+      var responsiveTabs = false;
+      var previousWindowWidth;
+      var tabWidths = [];
+      var tabHeight;
+      var widestTabWidth = 0;
+      var expandControlWidth;
+      var activeTabNth = $('li.active', $primaryTabs).index();
+      var expandedTabsHeaderPadding = 0;
+      var defaultHeaderPadding = $('.l-header').css('padding-top');
+
+      // These are essentially breakpoints to be measured against the tabArea
+      var allTabsWidth;             // Will show all tabs
+      var activeTabAndBeforeWidth;  // Will chop off tabs after active tab
+      var activeTabAndAfterWidth;   // Will chop off tabs before active tab
+
+      function initResponsivePrimaryTabs() {
+        tabHeight = $('li:first-child', $primaryTabs).outerHeight();
+        $($tabsWrapper).once('responsive-tabs', function(){
+          $($primaryTabs).after(
+            '<div class="expand-dropdown-tabs-control" aria-hidden="true" style="height: ' + tabHeight + 'px">' +
+              '<span class="expand-dropdown-tabs-label"></span>' +
+            '</div>'
+          );
+          $('.expand-dropdown-tabs-control', $tabsWrapper).click(function(){
+            $($tabsWrapper).toggleClass('expand-dropdown-tabs');
+            $(this).toggleClass('js-active');
+            // If there's not enough room for mobile tabs
+            if (expandedTabsHeaderPadding > 0 && $($tabsWrapper).hasClass('expand-dropdown-tabs')) {
+              $('.l-header').css('padding-top', expandedTabsHeaderPadding + 'px');
+              $('body').scrollTop($('body').scrollTop() + expandedTabsHeaderPadding - parseInt(defaultHeaderPadding, 10));
+            } else {
+              $('.l-header').css('padding-top', defaultHeaderPadding);
```

Comment:
Do you think this padding could be added to an element that is not layout-dependent? Maybe an extra element could be added to the DOM just before the tabs to provide the extra padding? Then remove this element when the tabs are closed.